File: packages/flutter/lib/src/echo_butler.dart
import 'package:http/http.dart' as http;
import 'mood/mood_client.dart';
import 'stellar/stellar_client.dart';
import 'social/social_client.dart';

/// Main entry point for the EchoButler SDK.
///
/// Call [EchoButler.initialize] once at app startup, then access
/// sub-clients via [EchoButler.mood], [EchoButler.stellar], etc.
class EchoButler {
  EchoButler._();

  static EchoButler? _instance;

  late final MoodClient mood;
  late final StellarClient stellar;
  late final SocialClient social;

  static EchoButler get instance {
    assert(_instance != null, 'Call EchoButler.initialize() first.');
    return _instance!;
  }

  /// Initialize the SDK. Call once in main() before runApp().
  ///
  /// ```dart
  /// await EchoButler.initialize(
  ///   apiKey: 'your_api_key',
  ///   network: StellarNetwork.testnet,
  /// );
  /// ```
  static Future<void> initialize({
    required String apiKey,
    String baseUrl = 'https://api.echobutler.dev/v1',
    StellarNetwork network = StellarNetwork.mainnet,
    http.Client? httpClient,
  }) async {
    final client = httpClient ?? http.Client();
    final config = _SDKConfig(
      apiKey: apiKey,
      baseUrl: baseUrl,
      network: network,
      httpClient: client,
    );

    _instance = EchoButler._()
      ..mood = MoodClient(config)
      ..stellar = StellarClient(config)
      ..social = SocialClient(config);
  }

  /// Set the auth token for authenticated API calls.
  void setAuthToken(String? token) {
    mood.config.authToken = token;
    stellar.config.authToken = token;
    social.config.authToken = token;
  }
}

enum StellarNetwork { mainnet, testnet }

class _SDKConfig {
  final String apiKey;
  final String baseUrl;
  final StellarNetwork network;
  final http.Client httpClient;
  String? authToken;

  _SDKConfig({
    required this.apiKey,
    required this.baseUrl,
    required this.network,
    required this.httpClient,
  });
}

// Make config accessible to sub-clients
typedef SDKConfig = _SDKConfig;

File: packages/swift/EchoButlerSDK/Sources/EchoButlerSDK/EchoButler.swift
import Foundation
import EchoButlerFFI

public struct EchoButlerConfig: Sendable {
    public var apiKey: String
    public var baseURL: String?
    public var network: StellarNetwork

    public init(
        apiKey: String,
        baseURL: String? = nil,
        network: StellarNetwork = .mainnet
    ) {
        self.apiKey = apiKey
        self.baseURL = baseURL
        self.network = network
    }
}

public enum StellarNetwork: UInt8, Codable, Sendable {
    case mainnet = 0
    case testnet = 1
}

public final class EchoButler {
    public let mood: MoodClient
    public let stellar: StellarClient
    public let social: SocialClient

    public init(config: EchoButlerConfig) throws {
        self.mood = try MoodClient(config: config)
        self.stellar = try StellarClient(config: config)
        self.social = try SocialClient(config: config)
    }
}

public enum EchoButlerVersion {
    public static func current() -> String {
        guard let raw = echobutler_version() else {
            return "unknown"
        }
        defer { echobutler_free_string(raw) }
        return String(validatingUTF8: raw) ?? "unknown"
    }
}
